src/Avalonia.Controls/Primitives/AdornerLayer.cs

```diff
@@ -38,7 +37,7 @@ public class AdornerLayer : Canvas
         /// Defines the <see cref="DefaultFocusAdorner"/> property.
         /// </summary>
         public static readonly StyledProperty<ITemplate<Control>?> DefaultFocusAdornerProperty =
-            AvaloniaProperty.Register<Control, ITemplate<Control>?>(nameof(DefaultFocusAdorner));
+            AvaloniaProperty.Register<AdornerLayer, ITemplate<Control>?>(nameof(DefaultFocusAdorner));
```

Comment:
@maxkatz6 I don't know if this fix is correct

Reply:
Yes